### ISSUE TITLE
bug:? strange code colorization in nightly extension

### ISSUE THREAD

**Issue by zzjin** [golang/vscode-go#1478]
When try to test dlv-dap, I update vscode-go extension from stable to nightly, but found code colorization strange on vars and func names.
vacode version:
```
Version: 1.57.0-insider (user setup)
Commit: fe23c2041f848d6c2f6bd27875112a37bcc82f84
Date: 2021-05-06T05:16:25.564Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.21370
```

origin vscode-go:
version: `v0.24.2`
screenshot: 
![image](https://user-images.githubusercontent.com/679757/117270050-6732c600-ae8b-11eb-868e-14eb81afc06d.png)

nightly 
version:``
screenshot:
![image](https://user-images.githubusercontent.com/679757/117269772-28047500-ae8b-11eb-9510-df2b9c160bcd.png)

is this change of green color by design?

**Comment by hyangah** [comment]
Thanks for the report. Can you please share your settings (anything involving "gopls" and "editor.semanticHighlighting.*" and theme setup), and the gopls trace captured following [the instruction](https://github.com/golang/vscode-go/blob/master/docs/troubleshooting.md#collect-gopls-information)? I suspect gopls's [semantic token](https://github.com/golang/vscode-go/blob/master/docs/settings.md#uisemantictokens) is enabled automatically in the nightly.

**Comment by hyangah** [comment]
@stamblerre told me that semantic toke is enabled in Nightly by default. I verified it too.
@zzjin You can disable this feature explicitly by configuring in settings.json.

```
"gopls": {
  "ui.semanticTokens": false
}
```
But we are currently experimenting this new feature and looking for feedback. Do you think this color change is undesirable or distracting? What do you think?

**Comment by zzjin** [comment]
@hyangah thanks for response. setting `ui.semanticTokens` false rebase normal.

I use vscode's default dark theme and not set any other `semanticHighlighting` settings.

IMO, the new `semanticTokens` changes some 'same color for same mean' behavior: (just example)

1. import path string is in double quote, but last path is not as same color as prefix path. (as shown below)
2. funcs are yellow, but struct impl func are green:
  ![image](https://user-images.githubusercontent.com/679757/117394621-0eb20600-af29-11eb-94c3-49e8c92d4871.png)

**Comment by hyangah** [comment]
Yeah, we are still unclear how themes and vscode assign colors for tokens yet.

cc @pjweinb

**Comment by pjweinb** [comment]
The change in color for the last component of an import is intended. If the code says
import "a/b/c"
then uses of the package look like
c.fn()
That is, the coloring shows the namespace of the import.

**Comment by Hades32** [comment]
I actually love `ui.semanticTokens`, especially that packages now have a different color vs vars, but there are definitely a few weird colorings.
Adding to what was already mentioned above, `nil` now has the color of variables, when using the default Dark+ theme but on the other hand many color themes don't support the coloring of theme package name in the import

**Comment by sammy-hughes** [comment]
I guess I'll add a comment here that variables/arguments with a declared type with an underlying type that is some function signature will be colored as a variable and not as a callable (function or method). Variables/arguments with a type that is directly a function signature are colored correctly as a callable.
